analyzers/tests/SonarAnalyzer.Test/Trackers/ArgumentTrackerTest.cs

```diff
@@ -0,0 +1,1288 @@
+﻿/*
+* SonarAnalyzer for .NET
+* Copyright (C) 2015-2023 SonarSource SA
+* mailto: contact AT sonarsource DOT com
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this program; if not, write to the Free Software Foundation,
+* Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+*/
+
+using Microsoft.CodeAnalysis.Text;
+using SonarAnalyzer.Helpers.Trackers;
+using CS = Microsoft.CodeAnalysis.CSharp.Syntax;
+using VB = Microsoft.CodeAnalysis.VisualBasic.Syntax;
+
+namespace SonarAnalyzer.UnitTest.Trackers;
+
+[TestClass]
+public class ArgumentTrackerTest
+{
+    [TestMethod]
+    public void Method_SimpleArgument()
+    {
+        var snippet = """
+            System.IFormatProvider provider = null;
+            1.ToString($$provider);
+            """;
+        var (node, model) = ArgumentAndModelCS(WrapInMethodCS(snippet));
+
+        var argument = ArgumentDescriptor.MethodInvocation(KnownType.System_Int32, "ToString", "provider", 0);
```

Comment:
I find the name misleading. This is not an argument it's a descriptor. I think it would be better to name this either `argumentDescriptor` or `descriptor`.

This applies for the other tests as well.